feat: add path filtering

carishka-naidoo opened the pull request:
This pull request introduces configurable path exclusion filtering to both the Azure DevOps and GitHub repository ingestors, allowing users to specify glob patterns to exclude files or directories from ingestion. It also adds validation and logging for exclusion patterns, and centralizes path filtering logic in a new utility. The changes improve flexibility, debugging, and maintainability of the ingestion process.

carishka-naidoo committed:
fix: add path filtering

carishka-naidoo committed:
fix: yarn lock file

carishka-naidoo committed:
fix: prettier check

carishka-naidoo committed:
add changeset

SweetOBurritO commented on plugins/ai-assistant-common/src/utils/path-filter.ts:
```diff
@@ -0,0 +1,143 @@
+/**
+ * Default path exclusion patterns for common build artifacts and dependencies
+ */
+export const DEFAULT_PATH_EXCLUSIONS = [
```
I like the idea of this path filter function being in the common but i feel like these defaults shouldnt be here. Theyre a bit specific to a problem and not generic. 

Rather move these defaults to a constants file in the ingestor itself and then check the config and if thats null default to these

carishka-naidoo committed:
feat: add configurable path exclusion filtering to ado and github ingestors

SweetOBurritO committed:
Add path filtering to GitHub and Azure ingestors

Enhance GitHub and Azure ingestors to support path filtering, enabling the exclusion of specific files from ingestion.
